PWGCF/GenericFramework/Tasks/flowGenericFramework.cxx

```diff
@@ -278,163 +320,183 @@ struct GenericFramework {
       return;
     QAregistry.fill(HIST("globalTracks_PVTracks"), collision.multNTracksPV(), tracks.size());
+    QAregistry.fill(HIST("cent_nch"), tracks.size(), centrality);
     loadCorrections(bc.timestamp());
-    processData<kRaw>(collision, tracks, centrality);
+    processCollision<kRaw>(collision, tracks, centrality);
   }
   PROCESS_SWITCH(GenericFramework, processRawData, "Process analysis for derived data", true);
 
+  Filter dcaFilter = nabs(aod::track::dcaXY) < cfgDCAxy;
+  void processReco(soa::Filtered<soa::Join<aod::Collisions, aod::EvSels, aod::CentFT0Cs>>::iterator const& collision, aod::BCsWithTimestamps const&, soa::Filtered<soa::Join<aod::Tracks, aod::TracksExtra, aod::TracksDCA, aod::McTrackLabels>> const& tracks, aod::McParticles const&)
+  {
+    const auto centrality = collision.centFT0C();
+    auto bc = collision.bc_as<aod::BCsWithTimestamps>();
+    if (!collision.sel8())
+      return;
+    loadCorrections(bc.timestamp());
+    processCollision<kRaw>(collision, tracks, centrality);
+  }
+  PROCESS_SWITCH(GenericFramework, processReco, "Process analysis for MC reconstructed events", false);
+
   void processGen(soa::Filtered<soa::Join<aod::Collisions, aod::CentFT0Cs, aod::McCollisionLabels>>::iterator const& collision, aod::McParticles const& particles)
   {
     if (collision.has_mcCollision()) {
       float centrality = collision.centFT0C();
-      if (particles.size() < 1)
-        return;
-      if (centrality < centbinning.front() || centrality > centbinning.back())
-        return;
-      float vtxz = collision.posZ();
-      if (cfgFillQA) {
-        QAregistry.fill(HIST("vtxZ_gen"), vtxz);
-      }
-      fGFW->Clear();
-      float l_Random = fRndm->Rndm();
-
-      for (auto& particle : particles) {
-        if (!particle.isPhysicalPrimary() || particle.eta() < etalow || particle.eta() > etaup || particle.pt() < ptlow || particle.pt() > ptup)
-          continue;
-        bool WithinPtPOI = (ptpoilow < particle.pt()) && (particle.pt() < ptpoiup); // within POI pT range
-        bool WithinPtRef = (ptreflow < particle.pt()) && (particle.pt() < ptrefup); // within RF pT range
-        if (WithinPtRef)
-          fGFW->Fill(particle.eta(), fPtAxis->FindBin(particle.pt()) - 1, particle.phi(), 1, 1);
-        if (WithinPtPOI)
-          fGFW->Fill(particle.eta(), fPtAxis->FindBin(particle.pt()) - 1, particle.phi(), 1, 2);
-        if (WithinPtPOI && WithinPtRef)
-          fGFW->Fill(particle.eta(), fPtAxis->FindBin(particle.pt()) - 1, particle.phi(), 1, 4);
-        if (cfgFillQA) {
-          QAregistry.fill(HIST("phi_gen"), particle.phi());
-          QAregistry.fill(HIST("eta_gen"), particle.eta());
-          QAregistry.fill(HIST("pt_gen"), particle.pt());
-        }
-      }
-      for (uint l_ind = 0; l_ind < corrconfigs.size(); l_ind++) {
-        FillFC(fFC_gen, corrconfigs.at(l_ind), centrality, l_Random);
-      }
+      processCollision<kGen>(collision, particles, centrality);
```

Comment:
In here be aware that the function is subscribed to a `collision` and it gets the tracks associated to that collision but also is getting the full `McParticles` table which means that that full table is being processed for each collision

I will suggest to run a few tests locally with different combinations. With a single AO2D file from an MC production it should be able to test reconstructed data, reconstructed MC data, and generated MC. Once reasonable results are obtained the PR should proceed

In case you need any help we can plan for a chat